### PR TITLE
Fix dataset train/valid/test allocation

### PR DESCRIPTION
**Pull request by ebianchi**
Currently the validation set was getting set to the same size as the training set, instead of to the allocated fraction provided as `config.valid_fraction`.  This fix ensures the dataset splits are done according to the `config.train_fraction` / `config.valid_fraction` / `config.test_fraction` settings as expected.

### CODE REVIEW

**Review by mshalm — APPROVED** [review]
LGTM
